src/huggingface_hub/keras_mixin.py

```diff
@@ -178,6 +301,12 @@ def push_to_hub_keras(
         include_optimizer=include_optimizer,
         **model_save_kwargs,
     )
+
+    hyperparameters = _extract_hyperparameters_from_keras(model)
+    if model_plot and is_graphviz_available() and is_pydot_available():
+        _plot_network(model, repo_path_or_name)
+    lines = _parse_model_history(model)
```

Comment:
I feel all of these should be within `_create_model_card`. It doesn't make sense to compute before the function if you're not using this info outside.